most complete error checking i can come up with, added transactions, changed message output functionality

ben1215 opened the pull request:
Please test this as best as you can, but I think we are good to go!

etp12 commented:
When I tested this on the unixs.cis.pitt.edu server after selecting an option at the menu in the Driver (which uses .nextInt()), if you selected createUser() the first thing it asks you for is a string for the full name but after using nextInt() there is still a new line character in the buffer from pressing enter, so it would be skipped and the user wouldn't get a chance to enter a name. ![](https://i.gyazo.com/d3ad035836a70cc2aa61a1d382134117.png) This is what it looked like when I tested it. I used multiple Scanners in my first version to avoid this, but I could've also put a .nextLine() statement after reading the user's menu selection to consume the new line character which is what I just added.

I guess my editor changes some line edits which is why my commit looks like it changed a bunch of stuff, but I just added that one line so it would run on the unixs.cis.pitt.edu server correctly.

Other than that everything works fine for me and I'll leave it up to you to merge after you see this. I'll follow this format for the other functions as well.

ben1215 commented:
ohhhhhhh. My bad man. I must've missed that. I hope you understand, I only changed the way the scanners worked because the whole newline/buffer thing caused an issue with the try-catch i put it in to catch wrong datatype errors. If it was expecting a number and you put a letter 2 times in a row, it would go on an infinite loop. that's why i changed it. I must've just missed that. My bad, I thought i checked that. With your change, will it properly catch all the errors thown if you put in the wrong datatype multiple times in a row? and loop back through the program drive successfully?


- Ben Kristofic
On 4/14/2016 06:30:32 AM, Ethan Pavolik <notifications@github.com> wrote:
When I tested this on the unixs.cis.pitt.edu server after selecting an option at the menu in the Driver (which uses .nextInt()), if you selected createUser() the first thing it asks you for is a string for the full name but after using nextInt() there is still a new line character in the buffer from pressing enter, so it would be skipped and the user wouldn't get a chance to enter a name. [https://camo.githubusercontent.com/fd1ab77bce064225a7cfce976ca4405ead0b9540/68747470733a2f2f692e6779617a6f2e636f6d2f64336164303335383336613730636332616136316131643338323133343131372e706e67] This is what it looked like when I tested it. I used multiple Scan ners in my first version to avoid this, but I could've also put a .nextLine() statement after reading the user's menu selection to consume the new line character which is what I just added.
I guess my editor changes some line edits which is why my commit looks like it changed a bunch of stuff, but I just added that one line so it would run on the unixs.cis.pitt.edu server correctly.
Other than that everything works fine for me and I'll leave it up to you to merge after you see this. I'll follow this format for the other functions as well.
—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub [https://github.com/ben1215/FaceSpace/pull/4#issuecomment-209870150]
